core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -342,7 +355,9 @@ RefreshingAddressResolverGroup build(EventLoopGroup eventLoopGroup) {
                 builder.decodeIdn(decodeIdn);
             }
         };
+        final String cacheSpec = firstNonNull(this.cacheSpec, Flags.dnsCacheSpec());
```

Comment:
`Flags.dnsCacheSpec()` is annotated with `@Nullable`. Is it intended?

Reply:
It will be null when specified like: `-Dcom.linecorp.armeria.dnsCacheSpec=off`.
https://github.com/line/armeria/blob/6e0e47b60a67b8680843dae2513f051ff4f72076/core/src/main/java/com/linecorp/armeria/common/Flags.java#L1071-L1084

`Flags.dnsCacheSpec()` is itself nullable, but `firstNonNull` checks to not be null.
Would it better to make this `NonNull` when we define `dnsCacheSpec`?

Reply:
>  Would it better to make this NonNull when we define dnsCacheSpec?

+1 Because the error message of `firstNonNull(...)` would be ambiguous. Users can get a detailed error message early by making `dnsCacheSpec()` non-null.
https://github.com/google/guava/blob/7ef9f97cad23efee4cc18c63b5f3c90ca320f131/guava/src/com/google/common/base/MoreObjects.java#L64
```
"Both parameters are null"
```

Reply:
SGTM 👍